world.go

```diff
@@ -296,7 +339,12 @@ func (w *World) updateComponentFlags(id EID) {
 
 		// ... for the bit index that corresponds to the map id,
 		// use the true/false value we just obtained as the value for that bit in the bitset.
-		w.ComponentFlags[id].Set(int(w.factories[idx].ID()), found)
+		cf, cfFound := w.ComponentFlags.Load(id)
+		if !cfFound {
+			continue
+		}
```

Comment:
We almost certainly should explode if the component flags aren't found. We need to write more tests that hit this and really think about what it means for us to wind up here, and what we should do about it.